emmet-core/emmet/core/qc_tasks.py

```diff
@@ -0,0 +1,609 @@
+# mypy: ignore-errors
+
+""" Core definition of a Q-Chem Task Document """
+from typing import Any, Dict, List, Optional
+import logging
+import os
+import re
+from collections import OrderedDict
+from pydantic import BaseModel, Field
+from pymatgen.core.structure import Molecule
+from pymatgen.io.qchem.inputs import QCInput
+from monty.serialization import loadfn
+from typing import Type, TypeVar, Union
+from emmet.core.structure import MoleculeMetadata
+from pathlib import Path
+from emmet.core.qchem.calc_types import (
+    LevelOfTheory,
+    CalcType,
+    TaskType,
+)
+from emmet.core.qchem.calculation import Calculation
+
+from emmet.core.qchem.task import QChemStatus
+
+
+__author__ = (
+    "Evan Spotte-Smith <ewcspottesmith@lbl.gov>, Rishabh D. Guha <rdguha@lbl.gov>"
+)
+
+logger = logging.getLogger(__name__)
+_T = TypeVar("_T", bound="TaskDoc")
+# _DERIVATIVE_FILES = ("GRAD", "HESS")
+
+
+# class QChemStatus(ValueEnum):
+#     """
+#     Q-Chem Calculation State
+#     """
+
+#     SUCCESS = "successful"
+#     FAILED = "unsuccessful"
+
+
+class OutputDoc(BaseModel):
+    molecule: Molecule = Field(
+        None,
+        title="Output Molecule",
+        description="Output molecule after the QChem Calculation",
+    )
+
+    initial_molecule: Molecule = Field(None, description="Input Molecule object")
+    optimized_molecule: Molecule = Field(None, description="Optimized Molecule object")
+
+    # TODO: Discuss with Evan if these go here
+    # species_hash: str = Field(
+    #     None,
+    #     description="Weisfeiler Lehman (WL) graph hash using the atom species as the graph node attribute.",
+    # )
+    # coord_hash: str = Field(
+    #     None,
+    #     description="Weisfeiler Lehman (WL) graph hash using the atom coordinates as the graph node attribute.",
+    # )
+
+    # last_updated: datetime = Field(
+    #     None,
+    #     description = "Timestamp for the most recent calculation for this QChem task document",
+    # )
+
+    final_energy: float = Field(
```

Comment:
This is probably a fine start, but it would be nice to have more output data on the top level. Currently, most of the data is stuck in calcs_reversed; that's fine (at least it's somewhere and accessible to the user), but makes accessing it a bit annoying.